src/ngx_http_lua_subrequest.c

```diff
@@ -382,15 +404,62 @@ ngx_http_lua_ngx_location_capture_multi(lua_State *L)
 
             type = lua_type(L, -1);
 
-            if (type == LUA_TNIL) {
-                method = NGX_HTTP_GET;
+            switch (type) {
+                case LUA_TNIL:
+                    method = NGX_HTTP_GET;
+                    break;
+                case LUA_TNUMBER:
+                    method = (ngx_uint_t) lua_tonumber(L, -1);
+                    break;
+                case LUA_TSTRING:
+                    lua_method_name = (u_char *) lua_tolstring(L, -1,
+                                                        &method_name_length);
+                    if (method_name_length == 0) {
+                        return luaL_error(L, "Bad http request method");
+                    }
+                    
+                    methods_number = sizeof(ngx_http_lua_ordered_methods) /
+                        sizeof(ngx_http_lua_ordered_methods[0]);
+                    for (method_index = 0; method_index < methods_number;
+                         method_index++) {
+                        if (ngx_strncasecmp(
+                              ngx_http_lua_ordered_methods[method_index]->data,
+                              lua_method_name,
+                              method_name_length)
+                            == 0) {
```

Comment:
Style: 1. we should align the first argument on each continued lines with `(` of the function call on the first line, and 2. we should put the final `{` on its own line.